### PR TITLE
New attempt to build on Windows

### PR DESCRIPTION
**Pull request by metab0t**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Another attempt to build on Windows based on https://github.com/conda-forge/python-duckdb-feedstock/pull/42 and I have tried it on my local machine.
The [utf8 patch](https://github.com/duckdb/duckdb/pull/5129) has been upstreamed.

@conda-forge-admin, please rerender

### CODE REVIEW

**Comment by metab0t** [comment]
`tools\pythonpkg\tests\slow\test_h2oai_arrow.py` is shaky and failed in some Windows pipelines (spotted MemoryError).

**Comment by metab0t** [comment]
@gforsyth @dhirschfeld Any suggestions?

**Comment by dhirschfeld** [comment]
If it's just a memory error on windows, maybe skip the offending test?

May have to patch the source to skip it.

**Comment by metab0t** [comment]
@conda-forge-admin, please rerender

**Comment by metab0t** [comment]
@conda-forge-admin, please rerender

**Comment by metab0t** [comment]
@conda-forge-admin, please rerender

File: recipe/setup_py.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+diff --git a/tools/pythonpkg/setup.py b/tools/pythonpkg/setup.py
+--- a/tools/pythonpkg/setup.py
++++ b/tools/pythonpkg/setup.py
+@@ -286,7 +286,7 @@ def setup_data_files(data_files):
+         'duckdb-stubs'
+     ],
+     include_package_data=True,
+-    setup_requires=setup_requires + ["setuptools_scm<7.0.0", 'numpy>=1.14', 'pybind11>=2.6.0'],
++    setup_requires=setup_requires + ["setuptools_scm", 'numpy>=1.14', 'pybind11>=2.6.0'],
```
Comment (gforsyth) [inline recipe/setup_py.patch:12]: if you update from the `main` branch you can skip this patch since I added `setuptools` to the `build` environment.

**Review by gforsyth — COMMENTED** [review]
Hey @metab0t -- few conflicts since we merged the 0.6.0 update if you want to pull those in here.

**Comment by gforsyth** [comment]
Ahh, ok, the failures on the OSX builds are due to extension lookup failure and I've relaxed those checks on `main`, so if you update and resolve the conflicts with `main` I think this is just about ready to go in!

**Comment by metab0t** [comment]
@gforsyth I have rebased it on main branch. CI is green now.

**Review by gforsyth — APPROVED** [review]
Woohoo! Thanks @metab0t !

**Pull request merged**